server/test/repository/CategoryRepositoryTest.java

```diff
@@ -0,0 +1,80 @@
+package repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+import models.CategoryModel;
+import org.junit.Before;
+import org.junit.Test;
+import play.i18n.Lang;
+
+public class CategoryRepositoryTest extends ResetPostgres {
+
+  private CategoryRepository repo;
+
+  @Before
+  public void setupCategoryRepository() {
+    repo = instanceOf(CategoryRepository.class);
+  }
+
+  @Test
+  public void fetchOrInsertUniqueCategory_insertsNewCategory() {
+    CategoryModel category = createCategory("Housing", "Vivienda");
+    CategoryModel insertedCategory = repo.fetchOrInsertUniqueCategory(category);
+
+    assertThat(insertedCategory.getId()).isNotNull();
+    assertThat(insertedCategory.getDefaultName()).isEqualTo("Housing");
+    assertThat(insertedCategory.getLocalizedName().hasTranslationFor(Locale.US)).isTrue();
+  }
+
+  @Test
+  public void listCategories_fetchesAllCategories() {
+    CategoryModel category1 = createCategory("Health", "Salud");
+    repo.fetchOrInsertUniqueCategory(category1);
+
+    CategoryModel category2 = createCategory("Education", "Educación");
+    repo.fetchOrInsertUniqueCategory(category2);
+
+    ImmutableList<CategoryModel> allCategories = repo.listCategories();
+
+    assertThat(allCategories).hasSize(2);
+  }
+
+  @Test
+  public void fetchOrInsertUniqueCategory_fetchesExistingCategory() {
```

Comment:
Should probably also assert that the total number of categories is 1 (a duplicate wasn't added) 

(or add a new test for that case, but I'm lazy in cases like this and would just assert :)

Reply:
Good idea!  I added the assertion.